### PR TITLE
Fix: Avoid unnecessary aliases

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] avoids unnecessary aliases

Related to #953.

### CODE REVIEW

File: classes/Http/Form/ForgotForm.php [hunk 1]
```diff
@@ -17,13 +17,13 @@
 use Symfony\Component\Form\Extension\Core\Type\EmailType;
 use Symfony\Component\Form\Extension\Core\Type\SubmitType;
 use Symfony\Component\Form\FormBuilderInterface;
-use Symfony\Component\Validator\Constraints as Assert;
+use Symfony\Component\Validator;
```
Comment (derrabus) [inline classes/Http/Form/ForgotForm.php]: It's actually a common practice to import Validator constraints with alias `Assert`. 😕

File: classes/Infrastructure/Auth/SentinelAccountManagement.php [hunk 2]
```diff
@@ -34,7 +34,7 @@ public function findById(int $userId): UserInterface
     {
         $user = $this->sentinel->getUserRepository()->findById($userId);
 
-        if ($user instanceof SentinelUserInterface) {
+        if ($user instanceof Users\UserInterface) {
```
Comment (derrabus) [inline classes/Infrastructure/Auth/SentinelAccountManagement.php]: The alias was actually more readable to me, tbh. 😕

File: classes/Infrastructure/Auth/SymfonySentinelSession.php [hunk 3]
```diff
@@ -13,13 +13,13 @@
 
 namespace OpenCFP\Infrastructure\Auth;
 
-use Cartalyst\Sentinel\Sessions\SessionInterface as SentinelSessionInterface;
-use Symfony\Component\HttpFoundation\Session\SessionInterface as SymfonySessionInterface;
+use Cartalyst\Sentinel\Sessions;
+use Symfony\Component\HttpFoundation;
```
Comment (derrabus) [inline classes/Infrastructure/Auth/SymfonySentinelSession.php]: In this file, we're adapting two different `SessionInterfaces`s. The aliases did improve readability here. 😕

**Comment by mdwheele** [comment]
I agree with @derrabus here. There's nothing "wrong" with using aliases. Those that are there to resolve naming conflicts are arguably candidates for this type of change, but even still, a lot of them could be argued as aliases for readability anyway.

Not of the same class of alias, but consider the Illuminate database manager. If I did not alias that, I would be type-hinting `Manager` in my infrastructure classes rather than `Database`, which is certainly more indicative of what's going on.

I'm :-1: for sweeping change that removes all aliases.

**Comment by localheinz** [comment]
@mdwheele 

What's the advantage of using aliases when they aren't needed?

**Comment by mdwheele** [comment]
> What's the advantage of using aliases when they aren't needed?

When they aren't needed, they aren't needed. If they arguably improve readability (`SentinelSessionInterface` vs `SymfonySessionInterface`) or are a idiomatic / common use of a library (`Validator\Constraints as Assert`), I think they are nice to have.

**Comment by localheinz** [comment]
@mdwheele 

> When they aren't needed, they aren't needed. If they arguably improve readability (SentinelSessionInterface vs SymfonySessionInterface) or are am idiomatic / common use of a library (Validator\Constraints as Assert), I think they are nice to have.

I don't understand the point you are making here, can you explain?

File: classes/Infrastructure/Auth/SentinelAccountManagement.php (hunk 2 above)
Comment (localheinz) [reply classes/Infrastructure/Auth/SentinelAccountManagement.php]: How is this more readable?

**Comment by mdwheele** [comment]
> I don't understand the point you are making here, can you explain?

* If an alias truly is not needed, then it is justifiably a candidate for removal.
* However...
  * If it subjectively improves readability, then that is an argument to keep the alias (as exemplified by [this comment](https://github.com/opencfp/opencfp/pull/954#discussion_r157341731)) 
  * If it is common / community-standard to alias particular classes for readability (as exemplified by [this comment](https://github.com/opencfp/opencfp/pull/954#discussion_r157341716) then that is an argument to keep the alias.

I do not think that aliases make things more complicated as a blanket judgement. I think they are a tool that makes sense to use sometimes and not others. There are several changes in this PR that would be positive use-cases for alias and it seems we're blanket removing all aliases on the premise that they are unecessary. PHP, itself, is technically unnecessary. I could be writing C or any other lower-level implementation. However, PHP improves readability.

**Pull request closed without merging**